osbrain/tests/test_agent_sync_publications.py

```diff
@@ -73,6 +74,152 @@ def on_error(agent):
     agent.error_log.append('error')
 
 
+@pytest.mark.parametrize('socket_type', ['PUB', 'SYNC_PUB'])
+def test_unsubscribe(nsproxy, socket_type):
+    '''
+    Test that unsubscribing from topics works fine.
+
+    The server will be publishing the natural numbers, in ascending order,
+    starting at one. Odd numbers will be sent with `odd` topic and even
+    numbers with `even` topic.
+
+    A subscriber will subscribe to all numbers, and then it will unsubscribe
+    from the odd numbers.
+
+    Two non consecutive numbers must be received consecutively at that point,
+    since they both will be even.
+    '''
+    def check_non_consecutive(agent):
+        """
+        Check whether the last two received numbers are non-consecutive.
+        """
+        if len(agent.received) < 2:
+            return False
+        return agent.received[-1] - agent.received[-2] != 1
+
+    def publish(agent):
+        assert agent.count % 2 == 1
+        agent.send('pub', agent.count, topic='odd')
+        agent.send('pub', agent.count + 1, topic='even')
+        agent.count += 2
+
+    # Set up the agents
+    server = run_agent('server')
+    client = run_agent('client', base=Client)
+
+    server.set_attr(count=1)
+    addr = server.bind(socket_type, alias='pub', handler=lambda: None)
+    client.connect(addr, alias='sub', handler={'odd': append_received,
+                                               'even': append_received})
+
+    # Make sure the connection is stablished
+    server.each(0.1, 'send', 'pub', 'connecting...', topic='odd', alias='tmp')
+    assert wait_agent_attr(client, data='connecting...')
+    server.stop_timer('tmp')
+
+    # Wait until client receives some numbers
+    client.set_attr(received=[])
+    server.each(0.1, publish)
+    assert wait_agent_attr(client, length=5)
```

Comment:
Yes, maybe it would be easier to read and understand the test. Maybe we can change it towards the end of the PR? To get the most important things tackled first.